python/src/otel/otel_sdk/otel-instrument

```diff
@@ -1,44 +1,160 @@
 #!/usr/bin/env python3
-# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# SPDX-License-Identifier: MIT-0
 
-from os import environ, system
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+`otel-instrument`
+
+This script configures and sets up OpenTelemetry Python with the values we
+expect will be used by the common user. It does this by setting the environment
+variables OpenTelemetry uses, and then initializing OpenTelemetry using the
+`opentelemetry-instrument` auto instrumentation script from the
+`opentelemetry-instrumentation` package.
+
+Additionally, this configuration the user is using packages derived from
+`opentelemetry-instrumentation` and `opentelemetry-sdk` in their implementation.
+
+Usage
+-----
+In the configuration of an AWS Lambda function with
+`otel-instrument` at the root level of a Lambda
+Layer:
+
+.. code::
+
+    AWS_LAMBDA_EXEC_WRAPPER = /opt/otel-instrument
+
+"""
+
+import os
 import sys
+from os import environ, execl
 
-# the path to the interpreter and all of the originally intended arguments
-args = sys.argv[1:]
+AWS_LAMBDA_FUNCTION_NAME = "AWS_LAMBDA_FUNCTION_NAME"
+PYTHONPATH = "PYTHONPATH"
+LAMBDA_RUNTIME_DIR = "LAMBDA_RUNTIME_DIR"
 
-# enable OTel wrapper
-environ["ORIG_HANDLER"] = environ.get("_HANDLER")
-environ["_HANDLER"] = "otel_wrapper.lambda_handler"
+# Update the python paths for packages with `sys.path` and `PYTHONPATH`
+
+# - Expect this `otel-instrument` file to be at the Lambda Layer root. Then, we
+#   know where the OpenTelemetry Python packages are and can add them to the
+#   PYTHONPATH.
+
+LAMBDA_LAYER_PKGS_DIR = os.path.abspath(
```

Comment:
I think this file would be located at `/var/task` - shouldn't this explicitly reference `/opt/python` which while not an environment variable is a well defined path for lambda?